### PR TITLE
Boilerplate spec

### PR DESCRIPTION
**Pull request by m0mus**

### CODE REVIEW

**Comment by m0mus** [comment]
@lukasj I used a template provided by @bshannon (see here: https://github.com/bshannon/spec-template). You better forward your question to Bill. :)

**Comment by bshannon** [comment]
@lukasj Is there a document that describes "where it belongs"?
I just put it near the pom.xml since it seemed to me to be just an
extension of the pom.xml as opposed to source code for the project.

**Comment by lukasj** [comment]
@bshannon see https://maven.apache.org/guides/mini/guide-assemblies.html which says:

> You'll notice that the assembly descriptor is located in ${project.basedir}/src/assembly which is the standard location for assembly descriptors.

and points to https://maven.apache.org/guides/introduction/introduction-to-the-standard-directory-layout.html

... I prefer `${project.basedir}/src/main/assembly` because IDEs treat each folder under `src/main` as a source root and makes it visible in project view as such by default and what I'm describing is actually an output based on project's main sources. In case I'll be doing test distribution, I put corresponding assembly descriptor under `src/test/assembly`.

In any case, as long as the descriptor is somewhere under `src` folder, I'm fine with whatever location you prefer.

**Comment by bshannon** [comment]
I'm fine with moving the assembly file to src/assembly/ if you think that's better.
I don't see the point of moving it from one non-standard location to another
non-standard location.  Again, to me, it just seemed like an extension of the pom.xml,
something related to building the project, not source code of the project itself, which
is why I put it in the top level directory.  But I'm all for standards so move it to
src/assembly/ if you want.

File: spec/pom.xml [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+                <executions>
+                    <execution>
+                        <id>enforce-versions</id>
+                        <goals>
+                            <goal>enforce</goal>
+                        </goals>
+                        <configuration>
+                            <rules>
+                                <requireJavaVersion>
+                                    <version>[1.8.0,1.9.0)</version>
```
Comment (Tomas-Kraus) [inline spec/pom.xml]: Why to enforce Java 8 here? It makes build a bit messy. API build requires Java 9 and later, spec Java 8. :(

**Pull request merged**